src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -538,7 +538,10 @@ CreateDistributedTable(Oid relationId, Var *distributionColumn, char distributio
 		Oid partitionRelationId = InvalidOid;
 		foreach_oid(partitionRelationId, partitionList)
 		{
-			CreateDistributedTable(partitionRelationId, distributionColumn,
+			Var *partitionDistributionColumn =
```

Comment:
a nicer approach could have been to use `columnName` in pg_dist_partition. But, (a) that's a breaking change (b) we want to backport this patch at least to 10.0 and could be very challenging to do that

Reply:
I'll open an issue to fix this later, in a major/breaking release

Reply:
makes sense